core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -384,15 +385,15 @@ class KafkaApis(val requestChannel: RequestChannel,
   /**
    * Handle a produce request
    */
-  def handleProduceRequest(request: RequestChannel.Request, requestLocal: RequestLocal): Unit = {
+  def handleProduceRequest(request: RequestChannel.Request, requestLocal: RequestLocal): CompletableFuture[Unit] = {
     val produceRequest = request.body[ProduceRequest]
 
     if (RequestUtils.hasTransactionalRecords(produceRequest)) {
       val isAuthorizedTransactional = produceRequest.transactionalId != null &&
         authHelper.authorize(request.context, WRITE, TRANSACTIONAL_ID, produceRequest.transactionalId)
       if (!isAuthorizedTransactional) {
         requestHelper.sendErrorResponseMaybeThrottle(request, Errors.TRANSACTIONAL_ID_AUTHORIZATION_FAILED.exception)
-        return
+        CompletableFuture.completedFuture[Unit](())
```

Comment:
I think we should not remove the `return`, WDYT?
```suggestion
        retrun CompletableFuture.completedFuture[Unit](())
```

Reply:
You’re right, this should have preserved the original early return behavior.
I’ve fixed it.